### DISCUSSION TITLE
parameterised queries. What syntax should be used?

### DISCUSSION THREAD

**Discussion by thislarz** [WolfgangFahl/pyLoDStorage#114] [Ideas]
Candidates:
- `$Var$` does not appear in SQL outside of strings.
- `utf-8` characters e.g «Var» no conflics with asci characters
- `<Var>` compatible to Backus-Naur notation
- `<?Var?>` similar to php
- `{{Var}}` used similarly in scholia and Jinja2 see https://stackoverflow.com/a/4181605/1497139

Should escape characters frame variable name on each side?

```python
        my_value = "Hello World"
        query = open(str(pathlib.Path(__file__).parent)+"/resources/queries/getEventsBySeriesId.sql").read()
        query.myvar = my_value
```

**Comment by WolfgangFahl** [comment]
see. https://github.com/WDscholia/scholia/tree/master/scholia/app/templates for how scholia does it

**Comment by WolfgangFahl** [comment]
see #113 
```python
qm=QueryManager(lang='sql',debug=False)
qm.queriesByName["getEventsBySeriesId"]
params={
   "SERIES_ID":seriesId
}
sql_query=query.getQuery(params)
listOfDicts=sqlDB.query(sql_query)
resultDoc=query.documentQueryResult(listOfDicts)   
```
